### PR TITLE
feat: add icons and tooltips to table header column prop

### PR DESCRIPTION
**Pull request by danidewitt**
## What did we change?
- Added new type to table column prop to allow the use of `EzIcon` and `EzTooltip`, with corresponding tests and documentation

## Why are we doing this?
- Designer request

## Screenshot(s) / Gif(s):
Screen Shot 2023-03-14 at 2 21 58 PM

## Checklist

- [x] Provide test coverage for the changes made
- [x] Create a changeset (`yarn changeset`) with a summary of the changes

[Contributing guidelines](https://ezcater.github.io/recipe/guides/contributing/)

### CODE REVIEW

**Review by pgrippi — APPROVED** [review]

File: packages/recipe/src/components/EzTable/EzTable.tsx [hunk 1]
```diff
@@ -229,8 +239,9 @@ const Thead = ({selectable}) => {
               sorted={isSorted(column)}
               onClick={event => onClick(event, column, sorting.onSortClick)}
             >
-              <span>
-                {heading} {sortable && <SortDirection direction={direction} />}
+              <span className={headerItems()}>
+                {heading} {icon && <span className={headerIcon()}>{icon}</span>}{' '}
```
Comment (pgrippi) [inline packages/recipe/src/components/EzTable/EzTable.tsx:243]: is the space (`{' '}`) needed with the `gap` property applied here? or is it just to give it some extra spacing?

Comment (danidewitt) [reply packages/recipe/src/components/EzTable/EzTable.tsx:243]: i didn't add it on purpose - it got auto-added on save

Comment (pgrippi) [reply packages/recipe/src/components/EzTable/EzTable.tsx:243]: ahhh I love when prettier does that!

**Review by noranda — CHANGES_REQUESTED** [review]
Do we need to support `EzIconButton` here? I only ask because I don't see it in the design and it does create a bit of UX confusion when paired with sortable headers, since you now have two separate onClicks happening in the same area. Would we be ok just supporting `EzIcon` with a tooltip?

**Comment by noranda** [comment]
Also, the `EzIconButton` and sort icon aren't showing up in the playroom (the opacity is `0`).
Screenshot 2023-03-16 at 10 23 01 AM

**Review by paul-ez — APPROVED** [review]

**Comment by danidewitt** [comment]
i added `EzIconButton` because it was in the ticket, but i'm definitely in favor of removing it

File: packages/recipe/src/components/EzTable/EzTable.md [hunk 2]
```diff
@@ -366,6 +366,70 @@ The following example performs client-side sorting, using the `key` metadata of
 };
 ```
 
+### Columns with Icons
+
+To add an icon to a column header, add `icon` to the `column`. Recipe supports `EzIcon`, which can be wrapped in `EzTooltip`.
+
+Consider and choose the appropriate the size, color, and accessibility of each icon added to table column headers in the context of the rest of the columns.
```
Comment (noranda) [inline packages/recipe/src/components/EzTable/EzTable.md:373]:
```suggestion
Consider and choose the appropriate size, color, and accessibility of each icon added to table column headers in the context of the rest of the columns.
```

**Review by noranda — APPROVED** [review]
Approved with typo fix. It might also be good to update the PR description here. Nice work!

**Review by pgrippi — APPROVED** [review]

**Review by paul-ez — APPROVED** [review]

**Pull request merged**